pkg/api/v1/control-plane/cache/generic_snapshot.go

```diff
@@ -103,17 +110,26 @@ func (s *GenericSnapshot) Consistent() error {
 
 	for _, ref := range required {
 		if resources, ok := s.typedResources[ref.Type]; ok {
-			if _, ok := resources.Items[ref.Name]; ok {
-				return fmt.Errorf("required resource not in snapshot: %s %s", ref.Type, ref.Name)
+			if _, ok := resources.Items[ref.Name]; !ok {
+				return fmt.Errorf("required resource name not in snapshot: %s %s", ref.Type, ref.Name)
 			}
 		} else {
-			return fmt.Errorf("required resource not in snapshot: %s %s", ref.Type, ref.Name)
+			return fmt.Errorf("required resource type not in snapshot: %s %s", ref.Type, ref.Name)
 		}
 	}
 
 	return nil
 }
 
```

Comment:
```suggestion
// MakeConsistent  should never be called on a generic snapshot as it is not used for snapshots with dependent resources.
```

Reply:
added comments for more clarity in https://github.com/solo-io/solo-kit/pull/462/commits/0859c25550a4c94a34e724db0b9f679805ef03bd